internal/kokoro/continuous.sh

```diff
@@ -0,0 +1,168 @@
+git clone . $GOCLOUD_HOME
+cd $GOCLOUD_HOME
+
+try3() { eval "$*" || eval "$*" || eval "$*"; }
+
+# All packages, including +build tools, are fetched.
+try3 go mod download
+./internal/kokoro/vet.sh
+
+# runDirectoryTests runs all tests in the current directory.
+# If a PATH argument is specified, it runs `go test [PATH]`.
+runDirectoryTests() {
+  go test -race -v -timeout 45m "${1:-./...}" 2>&1 \
+    | tee sponge_log.log
+  # Takes the kokoro output log (raw stdout) and creates a machine-parseable
+  # xUnit XML file.
+  cat sponge_log.log \
+    | go-junit-report -set-exit-code > sponge_log.xml
+  # Add the exit codes together so we exit non-zero if any module fails.
+  exit_code=$(($exit_code + $?))
+}
+
+# runEmulatorTests runs emulator tests in the current directory.
+runEmulatorTests() {
+  if [ -f "emulator_test.sh" ]; then
+    ./emulator_test.sh
+  fi
+  # Takes the kokoro output log (raw stdout) and creates a machine-parseable
+  # xUnit XML file.
+  cat sponge_log.log \
+    | go-junit-report -set-exit-code > sponge_log.xml
+  # Add the exit codes together so we exit non-zero if any module fails.
+  exit_code=$(($exit_code + $?))
+}
+
+# testAllModules runs all modules' tests, including emulator tests.
+testAllModules() {
+  echo "Testing all modules"
+  for i in $(find . -name go.mod); do
+    pushd "$(dirname "$i")" > /dev/null;
+      runDirectoryTests
+      # Run integration tests against an emulator.
+      runEmulatorTests
+    popd > /dev/null;
+  done
+}
+
+# testChangedModules runs tests in changed modules only.
+testChangedModules() {
+  for d in $CHANGED_DIRS; do
+    goDirectories="$(find "$d" -name "*.go" -printf "%h\n" | sort -u)"
+    if [[ -n "$goDirectories" ]]; then
+      for gd in $goDirectories; do
+        pushd "$gd" > /dev/null;
+          runDirectoryTests .
+        popd > /dev/null;
+      done
+    fi
+  done
+}
+
+set +e # Run all tests, don't stop after the first failure.
+exit_code=0
+
+if [[ $KOKORO_JOB_NAME == *"continuous"* ]]; then
+  # Continuous jobs only run root tests & tests in submodules changed by the PR.
+  SIGNIFICANT_CHANGES=$(git --no-pager diff --name-only $KOKORO_GIT_COMMIT^..$KOKORO_GIT_COMMIT | grep -Ev '(\.md$|^\.github)' || true)
+  # CHANGED_DIRS is the list of significant top-level directories that changed,
+  # but weren't deleted by the current PR. CHANGED_DIRS will be empty when run on main.
+  CHANGED_DIRS=$(echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | grep "/" | cut -d/ -f1 | sort -u | tr '\n' ' ' | xargs ls -d 2>/dev/null || true)
+  # If PR changes affect all submodules, then run all tests.
+  if [[ -z $SIGNIFICANT_CHANGES ]] || echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | grep "^go.mod$" || [[ $CHANGED_DIRS =~ "internal" ]]; then
+    testAllModules
+  else
+    runDirectoryTests . # Always run base tests.
+    echo "Running tests only in changed submodules: $CHANGED_DIRS"
+    testChangedModules
+  fi
+elif [[ $KOKORO_JOB_NAME == *"nightly"* ]]; then
```

Comment:
Do you have nightly set up, or just presubmit and continous?

Reply:
We only have continuous. I'll remove this elif clause.